vispy/app/backends/_qt.py

```diff
@@ -626,7 +652,17 @@ def _vispy_swap_buffers(self):
         # Swap front and back buffer
         if self._vispy_canvas is None:
             return
-        self.swapBuffers()
+        if QT5_NEW_API:
+            ctx = self.context()
+            ctx.swapBuffers(ctx.surface())
```

Comment:
I wonder if somehow repeated calls to these is problematic, you could try storing them in a private attribute of `self`